mlflow/shap.py

```diff
@@ -179,3 +255,338 @@ def predict_function(X) -> pred:
+                      describes model input and output :py:class:`Schema <mlflow.types.Schema>`.
+                      The model signature can be :py:func:`inferred <mlflow.models.infer_signature>`
+                      from datasets with valid model input (e.g. the training dataset with target
+                      column omitted) and valid model output (e.g. model predictions generated on
+                      the training dataset), for example:
+
+                      .. code-block:: python
+
+                        from mlflow.models.signature import infer_signature
+                        train = df.drop_column("target_label")
+                        predictions = ... # compute model predictions
+                        signature = infer_signature(train, predictions)
+    :param input_example: (Experimental) Input example provides one or several instances of valid
+                          model input. The example can be used as a hint of what data to feed the
+                          model. The given example will be converted to a Pandas DataFrame and then
+                          serialized to json using the Pandas split-oriented format. Bytes are
+                          base64-encoded.
+    :param await_registration_for: Number of seconds to wait for the model version to finish
+                            being created and is in ``READY`` status. By default, the function
+                            waits for five minutes. Specify 0 or None to skip waiting.
+
+    """
+
+    Model.log(
+        artifact_path=artifact_path,
+        flavor=mlflow.shap,
+        explainer=explainer,
+        conda_env=conda_env,
+        serialize_model_using_mlflow=serialize_model_using_mlflow,
+        registered_model_name=registered_model_name,
+        signature=signature,
+        input_example=input_example,
+        await_registration_for=await_registration_for,
+    )
+
+
+@experimental
+def save_model(
+    explainer,
+    path,
+    serialize_model_using_mlflow=True,
+    conda_env=None,
+    mlflow_model=None,
+    signature: ModelSignature = None,
+    input_example: ModelInputExample = None,
+):
+    """
+    Save a SHAP explainer to a path on the local file system. Produces an MLflow Model
+    containing the following flavors:
+
+        - :py:mod:`mlflow.shap`
+        - :py:mod:`mlflow.pyfunc`
+
+    :param explainer: SHAP explainer to be saved.
+    :param path: Local path where the explainer is to be saved.
+    :param serialize_model_using_mlflow: When set to True, MLflow will extract the underlying
+                                         model and serialize it as a MLModel, otherwise it
+                                         uses SHAP's internal serialization. Defaults to True.
+                                         Currently MLFlow serialization is only supported for
+                                         models of 'sklearn' or 'pytorch' flavors.
```

Comment:
```suggestion
    :param serialize_model_using_mlflow: When set to True, MLflow will extract the underlying
                                         model and serialize it as an MLmodel, otherwise it
                                         uses SHAP's internal serialization. Defaults to True.
                                         Currently MLflow serialization is only supported for
                                         models of 'sklearn' or 'pytorch' flavors.
```

A couple nits but looks good. Can we insert the same parameter description in `log_explainer`?

Reply:
added param to log_explainer